packages/dev/core/src/Misc/fileTools.ts

```diff
@@ -762,6 +762,42 @@ export const RequestFile = (
     return fileRequest;
 };
 
+/**
+ * Reads the mime type from a URL, if available.
+ * @param url
+ * @returns
+ */
+export const GetMimeType = (url: string): string | undefined => {
+    const { match, type } = TestBase64DataUrl(url);
+    if (match) {
+        return type || undefined;
+    }
+
+    const lastDot = url.lastIndexOf(".");
+    if (lastDot === -1 || lastDot === url.length - 1) {
+        return undefined;
+    }

```

Comment:
Nit - actually redundant, as the default switch case will be triggered in both of those cases, but we can stay safe.